### PR TITLE
fix: describe authorization required for service object

### LINKED ISSUE #570: README should be updated with required permissions
#### Environment details

1. Specify the API at the beginning of the title. For example, "BigQuery: ...").
   General, Core, and Other are also allowed as types
2. OS type and version:
3. Java version: 11
4. logging version(s): 2.3.1

#### Steps to reproduce

  1. Deploy to Cloud Run
  2. See error: `java.lang.RuntimeException: java.util.concurrent.ExecutionException: com.google.cloud.logging.LoggingException: io.grpc.StatusRuntimeException: PERMISSION_DENIED: The caller does not have permission`

#### Code example

Call MonitoredResourceUtil.getResource() directly or through using LoggingHandler.addHandler(LOGGER, new LoggingHandler()); for example with java.util.logging.Logger.

```java
private final static Logger LOGGER = Logger.getLogger("HelloWorldApplication");
LoggingHandler.addHandler(LOGGER, new LoggingHandler());
LOGGER.log(new LogRecord(Level.FINEST, "my message!"));
```

#### Stack trace
```
java.lang.RuntimeException: java.util.concurrent.ExecutionException: com.google.cloud.logging.LoggingException: io.grpc.StatusRuntimeException: PERMISSION_DENIED: The caller does not have permission
	at com.google.cloud.logging.LoggingImpl.flush(LoggingImpl.java:717)
	at com.google.cloud.logging.LoggingImpl.write(LoggingImpl.java:699)
	at com.google.cloud.logging.LoggingHandler.publish(LoggingHandler.java:237)
	at java.logging/java.util.logging.Logger.log(Unknown Source)
	at com.example.helloworld.HelloworldApplication$HelloworldController.hello(HelloworldApplication.java:59)
```

The docs need to be updated to mention that the service account of the Google Cloud service (for example Cloud Run) needs to be updated to have role `Logs Writer`,`roles/logging.logWriter`.

### PR DESCRIPTION
**Pull request by minherz**
Adds description of the authorization requirements for a service object.
Fixes markdown warnings in the document.

Fixes #570

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,6 +11,7 @@ Java idiomatic client for [Cloud Logging][product-docs].
 ## Quickstart
 
 If you are using Maven with [BOM][libraries-bom], add this to your pom.xml file
+
```
Comment (simonz130) [inline README.md:14]: This file is autogenerated. See here: https://github.com/googleapis/java-logging/blob/master/.readme-partials.yaml where you need to make these changes so they are not overridden

**Review by simonz130 — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -81,17 +90,16 @@ to add `google-cloud-logging` as a dependency in your code.
...
 See the [Cloud Logging client library docs][javadocs] to learn how to
 use this Cloud Logging Client Library.
 
-
 #### Creating an authorized service object
-To make authenticated requests to Cloud Logging, you must create a service object with
-credentials. You can then make API calls by calling methods on the Logging service object. The
-simplest way to authenticate is to use
+
+To make requests to Cloud Logging, you must create a service object with valid credentials.
```
[the remark this answers is not in the document]
Comment (minherz) [reply README.md:100]: @simonz130 "service account" is not mentioned anywhere in the README. The "service object" is a class used in the code samples which credentials can belong to a user or to a service account.

File: README.md [hunk 3]
```diff
@@ -81,17 +90,16 @@ to add `google-cloud-logging` as a dependency in your code.
...
+You can then make API calls by calling methods on the Logging service object. The
+simplest way to get credentials is to use
```
[the remark this answers is not in the document]
Comment (minherz) [reply README.md:102]: OK

File: README.md (hunk 1 above)
Comment (minherz) [reply README.md:14]: I will update both places :+1:

Comment (minherz) [reply README.md:14]: I've found that an update to the first part of the README should also be applied to [java_library/README.md](https://github.com/googleapis/synthtool/blob/master/synthtool/gcp/templates/java_library/README.md) template. I will do it in a separate PR.

**Review by simonz130 — APPROVED** [review]

**Pull request merged**